INFO 2025-03-28 01:23:16,173 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 01:23:45,201 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\myweb\settings.py changed, reloading.
INFO 2025-03-28 01:23:53,164 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 01:49:49,671 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-28 01:49:50,107 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 16:28:00,301 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-28 16:28:00,943 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 16:50:12,791 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-28 16:50:13,244 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 17:45:55,066 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-28 17:45:55,543 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 22:16:52,000 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-28 22:16:52,627 autoreload Watching for file changes with StatReloader
INFO 2025-03-28 22:26:04,134 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\models.py changed, reloading.
INFO 2025-03-28 22:26:04,624 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:02:24,006 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:02:24,490 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:02:31,114 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:02:31,557 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:02:42,253 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:02:42,695 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:27:58,393 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:27:58,836 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:33:25,649 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:33:26,091 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:39:29,596 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:39:30,021 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:39:44,793 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:39:45,209 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:41:48,049 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\urls.py changed, reloading.
INFO 2025-03-29 01:41:48,486 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:42:10,580 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py changed, reloading.
INFO 2025-03-29 01:42:11,015 autoreload Watching for file changes with StatReloader
ERROR 2025-03-29 01:42:13,401 log Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: store_product.motor_control_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\views.py", line 46, in store
    logger.info(f"Store page rendered with {len(products)} products")
                                            ~~~^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: store_product.motor_control_id
ERROR 2025-03-29 01:42:13,412 basehttp "GET / HTTP/1.1" 500 130503
INFO 2025-03-29 01:50:53,058 autoreload D:\Desktop\PROJECTEND ระบบชำระเงินสมบูรแล้วเผื่อในกรณีโค้ดผิดพลาด4\myweb\store\models.py changed, reloading.
INFO 2025-03-29 01:50:53,539 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:53:12,935 autoreload Watching for file changes with StatReloader
INFO 2025-03-29 01:53:14,315 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-03-29 01:53:14,362 basehttp "GET /static/css/main.css HTTP/1.1" 304 0
INFO 2025-03-29 01:53:14,367 basehttp "GET /static/images/cart.png HTTP/1.1" 304 0
INFO 2025-03-29 01:53:14,367 basehttp "GET /images/shoes_yNyuZHS.jpg HTTP/1.1" 304 0
INFO 2025-03-29 01:53:14,390 basehttp "GET /images/poster_TUOkDrn.jpg HTTP/1.1" 304 0
INFO 2025-03-29 01:53:14,391 basehttp "GET /images/headphones_YycFlsb.jpg HTTP/1.1" 304 0
INFO 2025-03-29 01:53:14,394 basehttp "GET /static/js/cart.js HTTP/1.1" 304 0
INFO 2025-03-29 01:53:16,069 basehttp "GET /cart/ HTTP/1.1" 200 6116
INFO 2025-03-29 01:53:16,099 basehttp "GET /static/images/arrow-up.png HTTP/1.1" 304 0
INFO 2025-03-29 01:53:16,099 basehttp "GET /static/images/arrow-down.png HTTP/1.1" 304 0
INFO 2025-03-29 01:53:16,900 basehttp "GET /checkout/ HTTP/1.1" 200 9977
INFO 2025-03-29 01:53:21,181 basehttp "POST /process_order/ HTTP/1.1" 200 275
INFO 2025-03-29 01:55:54,230 autoreload Watching for file changes with StatReloader
INFO 2025-03-31 23:24:52,026 autoreload Watching for file changes with StatReloader
INFO 2025-03-31 23:24:53,487 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-03-31 23:24:54,706 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-03-31 23:24:55,442 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-03-31 23:24:59,284 basehttp "POST /process_order/ HTTP/1.1" 200 275
INFO 2025-03-31 23:27:40,841 basehttp "POST /process_order/ HTTP/1.1" 200 275
INFO 2025-03-31 23:33:46,092 autoreload Watching for file changes with StatReloader
INFO 2025-03-31 23:34:36,794 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-03-31 23:34:37,390 autoreload Watching for file changes with StatReloader
INFO 2025-03-31 23:44:21,574 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-03-31 23:53:57,092 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-03-31 23:59:00,065 autoreload Watching for file changes with StatReloader
INFO 2025-03-31 23:59:02,797 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-03-31 23:59:03,868 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-03-31 23:59:04,946 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-03-31 23:59:09,265 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 00:23:17,236 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-04-01 00:23:17,813 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:29:48,542 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\urls.py changed, reloading.
INFO 2025-04-01 00:29:49,250 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:34:27,784 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\urls.py changed, reloading.
INFO 2025-04-01 00:34:28,411 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:34:54,520 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-04-01 00:34:55,046 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:41:39,728 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-04-01 00:41:41,539 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-04-01 00:41:42,426 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-04-01 00:41:46,392 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 00:43:06,251 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:43:13,113 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 00:44:09,371 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 00:44:13,899 basehttp "POST /process_order/ HTTP/1.1" 200 276
WARNING 2025-04-01 00:51:10,581 log Not Found: /webhook/
WARNING 2025-04-01 00:51:10,582 basehttp "POST /webhook/ HTTP/1.1" 404 4809
INFO 2025-04-01 00:53:59,547 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\urls.py changed, reloading.
INFO 2025-04-01 00:54:00,133 autoreload Watching for file changes with StatReloader
WARNING 2025-04-01 00:54:31,293 log Bad Request: /webhook/
WARNING 2025-04-01 00:54:31,293 basehttp "POST /webhook/ HTTP/1.1" 400 38
ERROR 2025-04-01 00:54:47,617 log Internal Server Error: /webhook/opn
Traceback (most recent call last):
  File "D:\Desktop\ProjectEnd2\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "D:\Desktop\ProjectEnd2\venv\Lib\site-packages\django\utils\deprecation.py", line 131, in __call__
    response = self.process_response(request, response)
  File "D:\Desktop\ProjectEnd2\venv\Lib\site-packages\django\middleware\common.py", line 108, in process_response
    return self.response_redirect_class(self.get_full_path_with_slash(request))
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "D:\Desktop\ProjectEnd2\venv\Lib\site-packages\django\middleware\common.py", line 87, in get_full_path_with_slash
    raise RuntimeError(
    ...<9 lines>...
    )
RuntimeError: You called this URL via POST, but the URL doesn't end in a slash and you have APPEND_SLASH set. Django can't redirect to the slash URL while maintaining POST data. Change your form to point to 127.0.0.1:8000/webhook/opn/ (note the trailing slash), or set APPEND_SLASH=False in your Django settings.
ERROR 2025-04-01 00:54:47,620 basehttp "POST /webhook/opn HTTP/1.1" 500 70891
WARNING 2025-04-01 00:54:52,334 log Bad Request: /webhook/
WARNING 2025-04-01 00:54:52,335 basehttp "POST /webhook/ HTTP/1.1" 400 38
INFO 2025-04-01 01:00:53,631 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-04-01 01:00:54,177 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:01:13,046 autoreload Watching for file changes with StatReloader
ERROR 2025-04-01 01:01:19,680 log Internal Server Error: /webhook/
ERROR 2025-04-01 01:01:19,681 basehttp "POST /webhook/ HTTP/1.1" 500 38
INFO 2025-04-01 01:02:29,401 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:39:55,269 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:40:00,786 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-04-01 01:40:01,963 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-04-01 01:40:04,530 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-04-01 01:40:08,070 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 01:41:05,634 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:41:06,832 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-04-01 01:41:08,129 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-04-01 01:41:09,686 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-04-01 01:41:10,121 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-04-01 01:41:13,681 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 01:50:59,863 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-04-01 01:51:00,446 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:51:04,170 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:51:13,769 autoreload Watching for file changes with StatReloader
INFO 2025-04-01 01:51:15,039 basehttp "GET / HTTP/1.1" 200 5173
INFO 2025-04-01 01:51:16,099 basehttp "GET /cart/ HTTP/1.1" 200 4873
INFO 2025-04-01 01:51:16,691 basehttp "GET /checkout/ HTTP/1.1" 200 9311
INFO 2025-04-01 01:51:20,569 basehttp "POST /process_order/ HTTP/1.1" 200 276
INFO 2025-04-01 01:55:14,792 autoreload D:\Desktop\โปรเจค\PROJECTEND ระบบชำระเงินสมบูรแล้วติดปัญหาคุมมอเตอร์ - Copy\myweb\store\views.py changed, reloading.
INFO 2025-04-01 01:55:15,390 autoreload Watching for file changes with StatReloader
